[16.0] [MIG] maintenance_plan

alexeirivera87 opened the pull request.

grindtildeath committed:
[ADD] maintenance_plan module (#2)

pedrobaeza committed:
[FIX] maintenance_plan: README try me on runbot link

bestmazzo committed:
[MIG] maintenance_plan: Migration to 11.0

dalonsod committed:
[FIX] maintenance_plan: prevent preventive request cron crashing when equipment maintenance team is not filled

[ADD] icon.png

dalonsod committed:
[MIG] maintenance_plan: Migration to 12.0

[UPD] Update maintenance_plan.pot

AdriaGForgeFlow committed:
[12.0][IMP] maintenance_plan

[ADD] instructions and notes to Plan

- Add menus for maintenance plans.
- Set active button on maintenance plans.

improve navigation to maintenance plans.

[ADD] post-migration script

lromero-solvos committed:
Added translation using Weblate (Spanish)

lromero-solvos committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (30 of 30 strings)

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/es/

[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

dalonsod committed:
[FIX] maintenance_plan: cron generated request should inherit equipment company

maintenance_plan 12.0.1.1.0

dalonsod committed:
[FIX] maintenance_plan: next_maintenance_date should be stored

Commit:
[12.0][FIX] default equipment_id in plans

etobella committed:
[11.0][IMP] maintenance_plan: Return the created requests

etobella committed:
[12.0][FIX] maintenance_project_plan

etobella committed:
[12.0][IMP] maintenance_plan: Add team on plans

[UPD] Update maintenance_plan.pot

[UPD] README.rst

maintenance_plan 12.0.3.0.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/

dalonsod committed:
[FIX] maintenance_plan: prevent False as display name when name is empty

[UPD] Update maintenance_plan.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/

dalonsod committed:
Translated using Weblate (Spanish)

Currently translated at 98.4% (63 of 64 strings)

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/es/

NuriaXifre committed:
[IMP] maintenance_plan: black, isort

NuriaXifre committed:
[13.0][MIG] maintenance_plan

[UPD] Update maintenance_plan.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

Commit:
[12.0][IMP] maintenance_plan: no equipment

[UPD] Update maintenance_plan.pot

maintenance_plan 13.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

pre-commit update

AaronHForgeFlow committed:
[FIX]maintenance_plan migration scripts not applicable in v13

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

LoisRForgeFlow committed:
[13.0][IMP] maintenance_plan: make maintenance plan multi-company compatible.
includes migration scripts

LoisRForgeFlow committed:
maintenance_plan: manual version bump

[UPD] Update maintenance_plan.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

dalonsod committed:
[IMP] maintenance_plan: black, isort, prettier

dalonsod committed:
[MIG] maintenance_plan: Migration to 14.0

[UPD] Update maintenance_plan.pot

[UPD] README.rst

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_plan/it/

etobella committed:
[FIX] maintenance_plan: Remake the computation of next date
* Use latest request_date both for done and todo searches.
* When there is a latest done apply interval

Co-authored-by: Lois Rilo <lois.rilo@forgeflow.com>
Co-authored-by: LoisRForgeFlow <LoisRForgeFlow@users.noreply.github.com>

maintenance_plan 14.0.1.0.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_plan/it/

LoisRForgeFlow committed:
[IMP] maintenance_plan: next maintenance date tweaks

* only consider request after the start maintenance date.
* todo request ordered asc.
* add more tests.

maintenance_plan 14.0.1.0.2

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_plan/it/

BernatPForgeFlow committed:
[IMP] maintenance_plan: black, isort, prettier

BernatPForgeFlow committed:
[MIG] maintenance_plan: Migration to 15.0

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
[UPD] README.rst

mariadforgeflow committed:
[FIX] maintenance_plan: fix new request date when plan start date is later

OCA-git-bot committed:
maintenance_plan 15.0.1.0.1

MiquelRForgeFlow committed:
[FIX] maintenance_plan: avoid error if planning_step is not defined

Also, clean-up of useless from_string() calls.

mariadforgeflow committed:
[IMP] maintenance_plan: add chatter

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

OCA-git-bot committed:
maintenance_plan 15.0.1.1.1

sergiocorato committed:
[15.0][FIX] maintenance_plan generate requests for inactive equipment

OCA-git-bot committed:
maintenance_plan 15.0.1.1.2

mymage committed:
Translated using Weblate (Italian)

Currently translated at 67.9% (72 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (106 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
[REF] moving function get_relativedelta to method

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 58.4% (62 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/pt/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (106 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/pt/

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.2.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

victoralmau committed:
[IMP] maintenance_plan: Set planned_hours in request from plan if maintenance_timesheet is installed.

kluna1998 committed:
[FIX] maintenance_plan: fix maintence request form view

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

victoralmau committed:
[IMP] maintenance_plan: Add Generate requests for current threshold button from plan view form.

Add button from maintenance plan to allow generate maintenance requests
without waiting for cron.

TT41729

victoralmau committed:
[IMP] maintenance_plan: Add instructions to request report.

TT41671

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.3.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.4.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.5.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

OCA-git-bot committed:
maintenance_plan 15.0.1.5.1

victoralmau committed:
[IMP] maintenance_plan: Change tests to allow inheritance

TT41735

victoralmau committed:
[IMP] maintenance_plan: Skip assigned mail + Activity mail of the requests created from plans

TT41870

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.6.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

OCA-git-bot committed:
maintenance_plan 15.0.1.6.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

victoralmau committed:
[IMP] maintenance_plan: Add rules to maintenance.plan to view records of the equipments they follow

TT42535

OCA-git-bot committed:
maintenance_plan 15.0.1.7.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

etobella commented on maintenance_plan/tests/test_maintenance_plan.py:
```diff
@@ -228,10 +228,10 @@ def test_maintenance_request_report(self):
+        rendering, qweb_type = self.report_obj._render_qweb_text(
+            "base_maintenance.report_maintenance_request", generated_request.ids
+        )
+        self.assertFalse(rendering.decode("utf-8").find("TEST-INSTRUCTIONS") == -1)
```
REGEX was not working?

HekkiMelody commented on maintenance_plan/views/maintenance_equipment_views.xml:
```diff
@@ -0,0 +1,100 @@
+            <xpath expr="//group[@name='maintenance']" position="attributes">
+                <attribute name="attrs">{'invisible': 1}</attribute>
+            </xpath>
+            <xpath expr="//group[@name='maintenance']/.." position="after">
+                <field
+                    name="maintenance_plan_ids"
+                    nolabel="1"
+                    context="{'default_equipment_id': active_id, 'hide_equipment_id': 1}"
+                >
+                    <tree>
+                        <field name="maintenance_kind_id" string="Kind" />
+                        <field name="maintenance_team_id" string="Team" />
+                        <field name="name" />
+                        <field name="start_maintenance_date" string="Start Date" />
+                        <field name="interval" />
+                        <field name="interval_step" />
+                        <field name="duration" />
+                        <field name="next_maintenance_date" />
+                        <field
+                            name="maintenance_plan_horizon"
+                            string="P. Horizon period"
+                        />
+                        <field name="planning_step" string="P. Horizon step" />
+                    </tree>
+                </field>
+            </xpath>
```
Not sure why the change from 

`<xpath expr="//group[@name='maintenance']/" position="after">`

`<xpath expr="//group[@name='maintenance']/.." position="after">`

at the very least I would add a `<group string="Maintenance Plans"> ` around the whole thing, or alternatively remove the `nolabel` option

HekkiMelody reviewed (commented):
Overall looks good and seems to work, unsure about the reason for the few changes that were made compared to v15

HekkiMelody commented:
Also there was an existing proposed migration #324 , make sure to check if someone has done the work already, or explicitly point out that you're superseeding the previous PR, please!

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 85.9% (98 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/es/

mamcode commented:
> Also there was an existing proposed migration #324 , make sure to check if someone has done the work already, or explicitly point out that you're superseeding the previous PR, please!

There is also another duplicated PR that proposes to migrate this module to v16. #265

mcassuto commented:
@aleuffre @alexeirivera87 @mamcode 
Hi guys since there is 3 PR in total, what should we do now ?

HekkiMelody commented:
Leave a review on all of them, or the one you think is closest to being merged, both in terms of code and in terms of the author being more active.

In terms of code I think this one is closest to being done, there are just a few changes to be made.

Happy to open a 4th PR myself, since I'd like to have this merged ASAP :rofl:

mcassuto commented:
Ok let me check if it has been tested ok, I guess it is alreay done on production.

alexeirivera87 commented:
Hello @mcassuto  @aleuffre @mamcode 

Yes, I did check the existing PR before submitting this one:

About #324 , on this migration the only thing the author did was to change the module version (and even this is not correct), and to remove a function that was failing in the tests.

About #265 , here the only thing that was done was to change the version of the module.

Regards

HekkiMelody commented:
> Hello @mcassuto @aleuffre @mamcode
> 
> Yes, I did check the existing PR before submitting this one:
> 
> About #324 , on this migration the only thing the author did was to change the module version (and even this is not correct), and to remove a function that was failing in the tests.
> 
> About #265 , here the only thing that was done was to change the version of the module.
> 
> Regards

In that case, we usually write `superseds #number of the MR` in the description, so we can keep track of such things. Thank you :pray:

etobella commented:
Visually it works for me. Why did  you remove the Regex on the test?

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/es/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

HekkiMelody commented:
@alexeirivera87 if you could please address the changes that were requested, they're minor things and I think this is pretty close to getting merged.

if you don't have time, I'd like to pick it up myself (I'd preserve ownership of your commits) since I need this module merged as well.

Thank you for your contributions!

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

OCA-git-bot committed:
[UPD] README.rst

etobella commented:
@alexeirivera87 can you check the issue with the REGEX?

etobella committed:
[IMP] maintenance_plan: Allow to generate generic plans for multiple equipments defined with a domain

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/es/

etobella committed:
[FIX] maintenance_plan: The search must work with newIds

etobella committed:
[IMP] maintenance_plan: Add Team filter

Commit:
[UPD] Update maintenance_plan.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

victoralmau committed:
[IMP] maintenance_plan: Add freeze_time to prevent errors in tests

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[FIX] maintenance_plan: adapt tests

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (118 of 118 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

alexeirivera87 commented:
Hello @aleuffre @etobella 

I reverted the changes in unit tests, so assertRegex is used as before.

Now I have to check why the test failed in method test_next_maintenance_date_02

Regards

etobella commented:
But the test seems unrelated to the regex. Can you check it¿

alexeirivera87 commented:
@etobella All tests passed.

Yes, the is related to dates. Some are failing when current date is January 30th or 31st.

regards

etobella commented:
Wow!!! I see... that might happend every 4 years :laughing: 

One option would be to use freezetime in order to avoid this. Can you try?

victoralmau commented:
Hi, sorry for being late.
Added freeze_time in the tests in v15 https://github.com/OCA/maintenance/pull/371 but I see there are v15 changes missing in the commit history, can you add it?

The problem with the last days of the month is that the last day of the following month is not used (in those that are generated).

alexeirivera87 committed:
[IMP] maintenance_plan: pre-commit stuff

alexeirivera87 committed:
[MIG] maintenance_plan: Migration to 16.0

alexeirivera87 commented:
@victoralmau  @etobella 

It should be ok now.

regards

victoralmau reviewed (approved):
Code review OK

etobella commented:
/ocabot merge nobump

etobella commented:
/ocabot migraion maintenance_plan

etobella commented:
/ocabot migration maintenance_plan
